customerio/client_base.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import division
+from datetime import datetime
+import math
+import time
+
+from requests import Session
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+try:
+    from datetime import timezone
+    USE_PY3_TIMESTAMPS = True
+except ImportError:
+    USE_PY3_TIMESTAMPS = False
+
+
+class CustomerIOException(Exception):
+    pass
+
+
+class ClientBase(object):
+
+    def __init__(self, retries=3, timeout=10, backoff_factor=0.02):
+        self.timeout = timeout
+        self.retries = retries
+        self.setup_connection(retries, backoff_factor)
+
+    def setup_connection(self, retries, backoff_factor):
+        self.http = Session()
```

Comment:
The current approach to use one http session in the client's base and the children probably needs a revisit. 

If a customer is using both `APIClient` and `CustomerIO` (track) at the same time, it seems right now the same session will be used? Shouldn't each client use its own session given both handle it a bit differently (one uses siteid/key and the other uses bearer token)? Also, I think the server should be able to identify requests from these two clients as two separate sessions.